Mathlib/FieldTheory/Minpoly/IsConjRoot.lean

```diff
@@ -0,0 +1,310 @@
+`x` and `y` in `L`, `IsConjRoot K x y` is equivalent to the existence of an algebra equivalence
+`σ : L ≃ₐ[K] L` such that `y = σ x`.
+* `not_mem_iff_exists_ne_and_isConjRoot`: Let `L / K` be a field extension. If `x` is a separable
+element over `K` and the minimal polynomial of `x` splits in `L`, then `x` is not in the `K` iff
+there exists a different conjugate root of `x` in `L` over `K`.
+
+## TODO
+Prove `IsConjRoot.smul`, if `x` and `y` are conjugate roots, then so are `r • x` and `r • y`.
+
+## Tags
+conjugate root, minimal polynomial
+-/
+
+
+open Polynomial minpoly IntermediateField
+
+variable {R K L S A B: Type*} [CommRing R] [CommRing S] [Ring A] [Ring B] [Field K] [Field L]
+variable [Algebra R S] [Algebra R A] [Algebra R B]
+variable [Algebra K S] [Algebra K L] [Algebra K A] [Algebra L S]
+
+variable (R) in
+/--
+We say that `y` is a conjugate root of `x` over `K` if the minimal polynomial of `x` is the
+same as the minimal polynomial of `y`.
+-/
+def IsConjRoot (x y : A) : Prop := (minpoly R x) = (minpoly R y)
+
+/--
+The definition of conjugate roots.
+-/
+theorem isConjRoot_def {x y : A} : IsConjRoot R x y ↔ minpoly R x = minpoly R y := Iff.rfl
+
+namespace IsConjRoot
+
+/--
+Every element is a conjugate root of itself.
+-/
+theorem refl {x : A} : IsConjRoot R x x := rfl
+
+/--
+If `y` is a conjugate root of `x`, then `x` is also a conjugate root of `y`.
+-/
+theorem symm {x y : A} (h : IsConjRoot R x y) : IsConjRoot R y x := Eq.symm h
+
+/--
+If `y` is a conjugate root of `x` and `z` is a conjugate root of `y`, then `z` is a conjugate
+root of `x`.
+-/
+theorem trans {x y y': A} (h₁ : IsConjRoot R x y) (h₂ : IsConjRoot R y y') :
+    IsConjRoot R x y' := Eq.trans h₁ h₂
+
+/--
+Let `p` be the minimal polynomial of `x`. If `y` is a conjugate root of `x`, then `p y = 0`.
+-/
+theorem aeval_eq_zero {x y : A} (h : IsConjRoot R x y) : aeval y (minpoly R x) = 0 :=
+  h ▸ minpoly.aeval R y
+
+/--
+Let `r` be an element of the base ring. If `y` is a conjugate root of `x`, then `y + r` is a
+conjugate root of `x + r`.
+-/
+theorem add_algebraMap {x y : S} (r : K) (h : IsConjRoot K x y) :
+    IsConjRoot K (x + algebraMap K S r) (y + algebraMap K S r) := by
+  rw [isConjRoot_def, minpoly.add_algebraMap x r, minpoly.add_algebraMap y r, h]
+
+/--
+Let `r` be an element of the base ring. If `y` is a conjugate root of `x`, then `y - r` is a
+conjugate root of `x - r`.
+-/
+theorem sub_algebraMap {x y : S} (r : K) (h : IsConjRoot K x y) :
+    IsConjRoot K (x - algebraMap K S r) (y - algebraMap K S r) := by
+  simpa only [sub_eq_add_neg, map_neg] using add_algebraMap (-r) h
+
+/--
+If `y` is a conjugate root of `x`, then `-y` is a conjugate root of `-x`.
+-/
+theorem neg {x y : S} (h : IsConjRoot K x y) :
+    IsConjRoot K (-x) (-y) := by
+  rw [isConjRoot_def, minpoly.neg x, minpoly.neg y, h]
+
+end IsConjRoot
+
+open IsConjRoot
+
+variable [IsDomain S]
```

Comment:
Looks like this is only used much later. Can you move it down?

Reply:
Moved. Thank you!